spring-cloud-dataflow-registry/src/main/java/org/springframework/cloud/dataflow/registry/AppRegistry.java

```diff
@@ -122,95 +127,24 @@ public AppRegistration save(String name, ApplicationType type, URI uri, URI meta
 	}
 
 	public List<AppRegistration> importAll(boolean overwrite, Resource... resources) {
-
-		Set<String> keysAlreadyThere = overwrite ? Collections.emptySet() : uriRegistry.findAll().keySet();
-
-		List<AppRegistration> apps = new ArrayList<>();
-		for (Resource resource : resources) {
-			Properties properties = new Properties();
-			try (InputStream is = resource.getInputStream()) {
-				properties.load(is);
-
-				properties.entrySet().stream().map(toStringAndUri())
-						.flatMap(toValidAppRegistration(metadataUriFromProperties(properties)))
-						.filter(ar -> !keysAlreadyThere.contains(key(ar.getName(), ar.getType())))
-						.collect(Collectors.toList()) // Force eager evaluation to fail
-														// early
-						.forEach(ar -> apps.add(save(ar.getName(), ar.getType(), ar.getUri(), ar.getMetadataUri())));
-			}
-			catch (IOException e) {
-				throw new RuntimeException("Error reading from " + resource.getDescription(), e);
-			}
-		}
-
-		return apps;
-	}
-
-	private Function<Map.Entry<Object, Object>, AbstractMap.SimpleImmutableEntry<String, URI>> toStringAndUri() {
-		return kv -> {
-			try {
-				return new AbstractMap.SimpleImmutableEntry<>((String) kv.getKey(), new URI((String) kv.getValue()));
-			}
-			catch (URISyntaxException e) {
-				throw new IllegalArgumentException(e);
-			}
-		};
+		Set<String> registeredKeys = overwrite ? Collections.emptySet() : uriRegistry.findAll().keySet();
+		return Stream.of(resources)
+			.map(this::loadProperties)
+			.flatMap(prop -> prop.entrySet().stream()
+					.map(toStringAndUriFUNC)
+					.flatMap(kv -> toValidAppRegistration(kv, metadataUriFromProperties(kv.getKey(), prop)))
+					.filter(ar -> !registeredKeys.contains(key(ar.getName(), ar.getType())))
+					.map(ar -> save(ar.getName(), ar.getType(), ar.getUri(), ar.getMetadataUri()))
```

Comment:
i think the current behavior is to avoid saving any app registrations if a single one of them is already present.... verifying.  This PR would then change this behavior